### PR TITLE
Improvements to Quest event refactor PR

### PR DESCRIPTION
**Pull request by Laumesis**
I have not tested these enough for merge, but they most likely are still okey for that.

### CODE REVIEW

**Comment by Laumesis** [comment]
Note to Self: Last commits should be checked as I was half sleep.

`UNIT_FACTION` event is not good either as it fires too often. Has to look for better solution.

File: Modules/Quest/QuestEventHandler.lua [hunk 1]
```diff
@@ -70,6 +73,28 @@ function _QuestEventHandler:InitQuestLog()
         return false
     end
 
+    for i = 1, MAX_QUEST_LOG_INDEX do
```
Comment (BreakBB) [inline Modules/Quest/QuestEventHandler.lua:76]: This loop should be merged with the the loop below. Both are partly doing the same

File: Modules/Quest/QuestieQuest.lua [hunk 2]
```diff
@@ -444,7 +421,7 @@ function QuestieQuest:AbandonedQuest(questId)
 
         local quest = QuestieDB:GetQuest(questId);
         if quest then
-            quest.Objectives = {};
+            quest.Objectives = nil;
```
Comment (BreakBB) [inline Modules/Quest/QuestieQuest.lua:424]: I am not sure I like this change. I think it was just recently changed so a table, right? But an empty table is just as good as `nil` from the code logic perspective, but is also saver to not result in an error.

File: Modules/Quest/QuestEventHandler.lua (hunk 1 above)
Comment (BreakBB) [reply Modules/Quest/QuestEventHandler.lua:76]: We will keep these separate to make the code more readable. They are also run only on addon load so performance does not matter.

**Comment by Laumesis** [comment]
`UNIT_FACTION` seems okey to use if all reputation quest objectives require only standing change. can't be used for windsaber quests availability. I don't know where I saw it fire too much for `player` if I ever saw and not mixed it to something else.

TODO: init at login / reload should be tested better - specially with empty quest log.  And it could be implemented better.

**Comment by BreakBB** [comment]
Most of these changes are now already included in #3111

**Comment by Laumesis** [comment]
Closing this as all important changes are in #3111 now. I left few review comments.

Added fake debug level (to filter most spammy spam messages away) is not "merged" into #3111 and should be handled elsewhere in future.

**Pull request closed without merging**